packages/common/src/grpc-service.js

```diff
@@ -717,5 +733,19 @@ GrpcService.prototype.getService_ = function(protoOpts) {
   return service;
 };
 
+GrpcService.prototype.setGrpcMetadata_ = function(metadata) {
+  var cc = grpc.credentials.createFromMetadataGenerator(function(_, cb) {
+    var grpcMetadata = new grpc.Metadata();
+
+    for (var prop in metadata) {
+      grpcMetadata.add(prop, metadata[prop]);
+    }
```

Comment:
I would recommend creating and filling the metadata object outside of the metadata generator function. Some validation is done there, and it would be better to get the validation errors immediately, rather than waiting until a method is called.